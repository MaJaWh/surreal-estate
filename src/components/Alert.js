import React from 'react';

function Alert(props) {
  const { message, success } = props;
  return (
    <div className={`alert alert-${success ? 'success' : 'error'}`}>
      {message}
    </div>
  );
}

export default Alert;
