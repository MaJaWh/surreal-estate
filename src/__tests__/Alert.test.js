import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';

test('displays an error message', () => {
  const component = render(<Alert message="Error!" />);
  const alertMessageNode = component.getByText('Error!');

  expect(alertMessageNode.textContent).toBe('Error!');
});

test('displays a success message', () => {
  const { getByText } = render(<Alert message="Success!!!!" success />);

  expect(getByText(/Success/).textContent).toBe('Success!!!!');
});
