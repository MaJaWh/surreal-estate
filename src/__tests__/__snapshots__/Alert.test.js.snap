// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;

exports[`renders an error message when there is no response from the server 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
